File: design-patterns/go/builder/vehicle/car.go
// car.go

package main

import "fmt"

type Car struct {
	wheel    int
	engine   int
	seat     int
	door     int
	interior bool
}

func NewCar(noOfWheel int, noOfEngine int, noOfSeat int, noOfDoor int, interior bool) (car *Car) {
	car = &Car{}
	car.wheel = noOfWheel
	car.engine = noOfEngine
	car.seat = noOfSeat
	car.door = noOfDoor
	car.interior = interior
	return
}

func (car *Car) GetDoor() int {
	return car.door
}

func (car *Car) GetEngine() int {
	return car.engine
}

func (car *Car) GetSeat() int {
	return car.seat
}

func (car *Car) GetWheel() int {
	return car.wheel
}

func (car *Car) IsInterior() bool {
	return car.interior
}

func (car *Car) ToString() string {
	return fmt.Sprintf("%v%v%v%v%v%v%v%v%v%v", "Car: Wheel -> ", car.wheel, " | Engine -> ", car.engine, " | Seat -> ", car.seat, " | Door -> ", car.door, " | Interior -> ", car.interior)
}

File: design-patterns/ts/builder/vehicle/car.ts
// car.ts

class Car {
    private wheel: number;
    private engine: number;
    private seat: number;
    private door: number;
    private interior: boolean;

    constructor(noOfWheel: number, noOfEngine: number, noOfSeat: number, noOfDoor: number, interior: boolean) {
        this.wheel = noOfWheel;
        this.engine = noOfEngine;
        this.seat = noOfSeat;
        this.door = noOfDoor;
        this.interior = interior;
    }

    getWheel(): number {
        return this.wheel;
    }

    getEngine(): number {
        return this.engine;
    }

    getSeat(): number {
        return this.seat;
    }

    getDoor(): number {
        return this.door;
    }

    isInterior(): boolean {
        return this.interior;
    }

    toString(): string {
        return `Car:
        Wheel -> ${this.wheel}
        Engine -> ${this.engine}
        Seat -> ${this.seat}
        Door -> ${this.door}
        Interior -> ${this.interior}`;
    }
}

export default Car;